sdk/python/lib/pulumi/runtime/settings.py

```diff
@@ -125,12 +154,7 @@ def get_project() -> str:
     """
     Returns the current project name.
     """
-    project = SETTINGS.project
-    if not project:
-        raise RunError(
-            "Missing project name; for test mode, please call `pulumi.runtime.set_mocks`"
-        )
-    return project
+    return SETTINGS.project
 
 
 def _set_project(v: Optional[str]):
```

Comment:
It looks like `_set_project` and `_set_stack` are the only functions that modify the attributes of the global `SETTINGS` variable.

These were introduced in #2638 to support `PULUMI_TEST_MODE`, which was deprecated in #10400. Since this functionality was deprecated, I wonder if we can simply remove `_set_project` and `_set_stack`. It doesn't look like anything else in the repo is calling them.

Removing them may allow for a simpler implementation here, if we only make the top-level `SETTINGS` a `ContextVar`.